app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsPresenter.kt

```diff
@@ -1,13 +1,67 @@
 package ar.com.wolox.android.example.ui.home.news
 
+import ar.com.wolox.android.example.network.News
 import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import org.joda.time.DateTime
+import org.ocpsoft.prettytime.PrettyTime
 import javax.inject.Inject
 
-class NewsPresenter @Inject constructor() : BasePresenter<INewsView>() {
+class NewsPresenter @Inject constructor(private val newsAdapterAPI: NewsAdapterAPI) : BasePresenter<INewsView>() {
 
     override fun onViewAttached() {}
 
     fun onAddNewsButtonPressed() = view.goAddNews()
 
-    fun onPullDownRefresh() = view.nothingNewToShow()
+    /**
+     * This function doesn't compare
+     * older news with recent ones,
+     * for now.
+     */
+    fun onLoadOldNews() {
+        newsAdapterAPI.loadOlderNews({ onSuccessOlderNews(newsList = it) }, { onEmptyList() }, { onFailureOlderNews() })
+    }
+
+    private fun onSuccessOlderNews(newsList: List<News>) {
+        view.addOlderNews(
+                newsList.run {
```

Comment:
You don't need the `run` here. You should use `setReadableCreationTime(newsList)` directly.